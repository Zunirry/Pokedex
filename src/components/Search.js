import React, { useState, useEffect } from 'react';
import { 
    View,
    Text,
    Image,
    TextInput,
    Dimensions,
    StyleSheet
 } from 'react-native'
import loupe from '../../assets/loupe.png'
// import useSearch from './hook/useSearch'

const windowWidth = Dimensions.get('window').width;

const Search = ({data, handleSearch}) => {

    // const [ query, updateQuery] = useSearch()
    const [query, setQuery] = useState('')
    console.log(query, 'query');

    const updateQuery = (query) => {

        setQuery(query)

        const searchRes = data.filter(query => query)
            .filter(newQuery => newQuery.name.toLowerCase()
            .includes(query.toLowerCase()))

        console.log(searchRes, 'rsdadas');
        
        if(query == ''){
            handleSearch(data)
        }else{
            handleSearch(searchRes)
        }


        return searchRes
    }

    return (
        <View style={styles.container}>
            <Image
                source={loupe}
                resizeMode='contain'
                style={styles.img}
                tintColor='#b0b0b0'
            />

            <TextInput 
                placeholder='Search'
                style={styles.input}
                value={query}
                onChangeText={updateQuery}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    input: {
        backgroundColor: '#e0e0e0',
        paddingHorizontal: 10,
        marginVertical: 5,
        borderTopRightRadius: 10,
        borderBottomRightRadius: 10,
        width: windowWidth * 0.89
    },
    img: {
        borderRadius: 10,
        width: 25,
        height: 25,
        backgroundColor: '#e0e0e0',
        height: 49,
    }
})

export default Search;
