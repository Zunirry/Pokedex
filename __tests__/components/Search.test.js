import React from 'react';
import Search from '../../src/components/Search'
import { create } from 'react-test-renderer'

describe('<Search />', () => {

    test('Render of Search component ', () => {
        const search = create(<Search />).toJSON()
        expect(search).toMatchSnapshot();
    })
    
});



export default Search.test;
