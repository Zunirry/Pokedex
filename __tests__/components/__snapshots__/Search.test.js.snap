// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Render of Search component  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/loupe.png",
      }
    }
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 10,
        "height": 49,
        "width": 25,
      }
    }
    tintColor="#b0b0b0"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "borderBottomRightRadius": 10,
        "borderTopRightRadius": 10,
        "marginVertical": 5,
        "paddingHorizontal": 10,
        "width": 667.5,
      }
    }
    underlineColorAndroid="transparent"
    value={null}
  />
</View>
`;
